pkg/server/server.go

```diff
@@ -21,30 +22,62 @@ type Service interface {
 	Run(ctx context.Context, server *Server) error
 }
 
+type public struct {
+	listener net.Listener
+	grpc     *grpc.Server
+}
+
+type private struct {
+	listener net.Listener
+	grpc     *grpc.Server
+}
+
 // Server represents a bundle of services defined by a specific ID.
 // Examples of servers are the satellite, the storagenode, and the uplink.
 type Server struct {
-	lis      net.Listener
-	grpc     *grpc.Server
+	public   public
+	private  private
 	next     []Service
 	identity *identity.FullIdentity
 }
 
 // New creates a Server out of an Identity, a net.Listener,
 // a UnaryServerInterceptor, and a set of services.
-func New(opts *tlsopts.Options, lis net.Listener, interceptor grpc.UnaryServerInterceptor, services ...Service) (*Server, error) {
+func New(opts *tlsopts.Options, publicAddr, privateAddr string, interceptor grpc.UnaryServerInterceptor, services ...Service) (*Server, error) {
 	unaryInterceptor := unaryInterceptor
 	if interceptor != nil {
 		unaryInterceptor = combineInterceptors(unaryInterceptor, interceptor)
 	}
 
-	return &Server{
-		lis: lis,
+	publicListener, err := net.Listen("tcp", publicAddr)
+	if err != nil {
+		return nil, errs.Combine(err, publicListener.Close())
+	}
+	public := public{
+		listener: publicListener,
 		grpc: grpc.NewServer(
 			grpc.StreamInterceptor(streamInterceptor),
 			grpc.UnaryInterceptor(unaryInterceptor),
 			opts.ServerOption(),
 		),
+	}
+
+	privateListener, err := net.Listen("tcp", privateAddr)
+	if err != nil {
+		return nil, errs.Combine(err, privateListener.Close())
+	}
+	private := private{
+		listener: privateListener,
+		grpc: grpc.NewServer(
+			grpc.StreamInterceptor(streamInterceptor),
+			grpc.UnaryInterceptor(unaryInterceptor),
+			opts.ServerOption(),
```

Comment:
you probably don't need all of our security options for the private listener, since it is supposed to only listen on localhost anyway. there's a grpc insecure option to make it listen on plain http which could be good for local curl debugging

Reply:
good point. let me look into which options we need and remove the unnecessary ones